torch/csrc/stable/library.h

```diff
@@ -4,23 +4,80 @@
 
 #include <torch/csrc/inductor/aoti_torch/c/shim.h>
 
+#include <optional>
+
 // use anonymous namespace to avoid collisions between differing
 // versions of this file that may be included by different sources
 namespace {
 
-// helpers for converting between StableIValue and actual IValues
-template <typename T>
+namespace detail {
+// utility functions to detect optional
+template <typename V>
+struct is_optional : std::false_type {};
+template <typename V>
+struct is_optional<std::optional<V>> : std::true_type {};
+} // namespace detail
+
+template <
+    typename T,
+    std::enable_if_t<!detail::is_optional<T>::value, bool> = true>
 StableIValue from(T val) {
   static_assert(
       sizeof(T) <= sizeof(StableIValue),
       "StableLibrary stack does not support parameter types larger than 64 bits.");
   return *reinterpret_cast<StableIValue*>(&val);
 }
 
+// Specialization for std::nullopt_t
+template <>
+StableIValue from(std::nullopt_t val) {
+  return from(nullptr);
+}
+
+// Specialization for std::optional
 template <typename T>
+StableIValue from(std::optional<T> val) {
```

Comment:
What is T? Does this need to be a reference instead?

Reply:
no, we expect an rvalue to be passed in from the user